gpii/node_modules/gsettingsBridge/nodegsettings/nodegsettings.cc

```diff
@@ -15,119 +16,119 @@
   for (i = 0; keys[i]; i++) {
-    togo->Set(i,String::New(keys[i]));
+    togo->Set(i,String::NewFromUtf8(isolate, keys[i]));
   }
-  return scope.Close(togo);
+  info.GetReturnValue().Set(togo);
 }
 
-/* Should take schema and key */
-Handle<Value> get_gsetting(const Arguments& args) {
-  HandleScope scope;
+///* Should take schema and key */
+NAN_METHOD(get_gsetting) {
+  Isolate *isolate = info.GetIsolate();
   GSettings *settings;
-  char schema[1024];
-  char key[1024];
-  args[0]->ToString()->WriteAscii(schema);
-  settings = g_settings_new(schema);
-  args[1]->ToString()->WriteAscii(key);
+  Local<String> schema = info[0].As<String>();
+  Local<String> key = info[1].As<String>();
+  settings = g_settings_new(*String::Utf8Value(schema));
   GVariant* variant;
   const GVariantType* type;
 
-  variant = g_settings_get_value(settings,key);
+  variant = g_settings_get_value(settings,*String::Utf8Value(key));
   type = g_variant_get_type(variant);
 
   if (g_variant_type_equal(type,G_VARIANT_TYPE_DOUBLE)) {
-    return scope.Close(Number::New(g_variant_get_double(variant)));
+    info.GetReturnValue().Set(Number::New(isolate, g_variant_get_double(variant)));
   }
   else if (g_variant_type_equal(type,G_VARIANT_TYPE_INT32)) {
-    return scope.Close(Int32::New(g_variant_get_int32(variant)));
+    info.GetReturnValue().Set(Int32::New(isolate, g_variant_get_int32(variant)));
   }
   else if (g_variant_type_equal(type,G_VARIANT_TYPE_STRING)) {
-    return scope.Close(String::New(g_variant_get_string(variant,NULL)));
+    info.GetReturnValue().Set(String::NewFromUtf8(isolate, g_variant_get_string(variant,NULL)));
   }
   else if (g_variant_type_equal(type,G_VARIANT_TYPE_BOOLEAN)) {
-    return scope.Close(Boolean::New(g_variant_get_boolean(variant)));
+    info.GetReturnValue().Set(Boolean::New(isolate, g_variant_get_boolean(variant)));
   }
   else {
     g_print("The type is %s\n", g_variant_type_peek_string(type));
-    ThrowException(Exception::Error(String::New("Need to implement reading that value type")));
-    return scope.Close(Undefined());
+    ThrowError(String::NewFromUtf8(isolate,"Need to implement reading that value type"));
+    info.GetReturnValue().Set(Undefined(isolate));
   }
 }
 
-/* Should take schema, key, and value */
-Handle<Value> set_gsetting(const Arguments& args) {
-  HandleScope scope;
+///* Should take schema, key, and value */
+NAN_METHOD(set_gsetting) {
+  Isolate *isolate = info.GetIsolate();
   GSettings *settings;
   GVariant* variant;
   const GVariantType* type;
-  char schema[1024];
-  char key[1024];
+  Local<String> schema = info[0].As<String>();
+  Local<String> key = info[1].As<String>();
   bool status = false;
 
-  args[0]->ToString()->WriteAscii(schema);
-  settings = g_settings_new(schema);
-  args[1]->ToString()->WriteAscii(key);
-  if (args[2]->IsBoolean()) {
-    status = g_settings_set_boolean(settings,key,args[2]->BooleanValue());
+  settings = g_settings_new(*String::Utf8Value(schema));
```

Comment:
This would be a good opportunity to fix our long-standing bug due to never deallocating this object. By my reading - we should call g_object_unref, by I may misunderstand - http://stackoverflow.com/questions/2848273/should-a-g-object-new-have-a-matching-g-object-unref
What do you think?